ext/opentelemetry-ext-system-metrics/src/opentelemetry/ext/system_metrics/__init__.py

```diff
@@ -0,0 +1,198 @@
+    input("...")
+
+API
+---
+"""
+
+import gc
+import os
+import typing
+
+import psutil
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics.export import MetricsExporter
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+
+class SystemMetrics:
+    def __init__(
+        self,
+        exporter: MetricsExporter,
+        interval: int = 30,
+        labels: typing.Optional[typing.Dict[str, str]] = None,
+    ):
+        self._labels = {} if labels is None else labels
+        self.meter = metrics.get_meter(__name__)
+        self.controller = PushController(
+            meter=self.meter, exporter=exporter, interval=interval
+        )
+        self._proc = psutil.Process(os.getpid())
+        self._system_memory_labels = {}
+        self._system_cpu_labels = {}
+        self._network_bytes_labels = {}
+        self._runtime_memory_labels = {}
+        self._runtime_gc_labels = {}
+        # create the label set for each observer once
+        for key, value in self._labels.items():
+            self._system_memory_labels[key] = value
+            self._system_cpu_labels[key] = value
+            self._network_bytes_labels[key] = value
+            self._runtime_memory_labels[key] = value
+            self._runtime_gc_labels[key] = value
+
+        self.meter.register_observer(
+            callback=self._get_system_memory,
+            name="system.mem",
+            description="System memory",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_system_cpu,
+            name="system.cpu",
+            description="System CPU",
+            unit="seconds",
+            value_type=float,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_network_bytes,
+            name="system.net.bytes",
+            description="System network bytes",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_runtime_memory,
+            name="runtime.python.mem",
+            description="Runtime memory",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_runtime_gc_count,
+            name="runtime.python.gc.count",
+            description="Runtime: gc objects",
+            unit="objects",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+    def _get_system_memory(self, observer: metrics.Observer) -> None:
+        """Observer callback for memory available
+
+        Args:
+            observer: the observer to update
+        """
+        system_memory = psutil.virtual_memory()
+        _metrics = [
```

Comment:
I would define these outside of the callback. We don't want to keep instantiating these.

Reply:
I think we should add an API to modify the `_metrics` for each system metric. This way, users that use this library will be able to customize the types of metrics for each system metric they export.

Reply:
I agree, I've updated the constructor to support a `config` parameter to allow for other metrics to be configured. I think it can be improved, but maybe its enough to get started?